src/charm_state.py

```diff
@@ -0,0 +1,201 @@
+        }
+        try:
+            valid_flask_config = FlaskConfig(**flask_config)  # type: ignore
+        except ValidationError as exc:
+            error_fields = set(
+                itertools.chain.from_iterable(error["loc"] for error in exc.errors())
+            )
+            error_field_str = " ".join(f"flask_{f}" for f in error_fields)
+            raise CharmConfigInvalidError(f"invalid configuration: {error_field_str}") from exc
+        return cls(
+            flask_config=valid_flask_config.dict(exclude_unset=True, exclude_none=True),
+            webserver_workers=int(workers) if workers is not None else None,
+            webserver_threads=int(threads) if threads is not None else None,
+            webserver_keepalive=int(keepalive) if keepalive is not None else None,
+            webserver_timeout=int(timeout) if timeout is not None else None,
+        )
+
+    @property
+    def webserver_config(self) -> WebserverConfig:
+        """Get the web server configuration file content for the charm.
+
+        Returns:
+            The web server configuration file content for the charm.
+        """
+        return WebserverConfig(
+            workers=self._webserver_workers,
+            threads=self._webserver_threads,
+            keepalive=datetime.timedelta(seconds=int(self._webserver_keepalive))
+            if self._webserver_keepalive is not None
+            else None,
+            timeout=datetime.timedelta(seconds=int(self._webserver_timeout))
+            if self._webserver_timeout is not None
+            else None,
+        )
+
+    @property
+    def flask_config(self) -> dict[str, str | int | bool]:
+        """Get the value of the flask_config charm configuration.
+
+        Returns:
+            The value of the flask_config charm configuration.
+        """
+        return self._flask_config.copy()
+
+    @property
+    def base_dir(self) -> pathlib.Path:
+        """Get the base directory of the Flask application.
+
+        Returns:
+            The base directory of the Flask application.
+        """
+        return pathlib.Path("/srv/flask")
+
+    @property
+    def flask_dir(self) -> pathlib.Path:
+        """Gets the path to the Flask directory.
+
+        Returns:
+            The path to the Flask directory.
+        """
+        return self.base_dir / "app"
+
+    @property
+    def flask_wsgi_app_path(self) -> str:
+        """Gets the Flask WSGI application in pattern $(MODULE_NAME):$(VARIABLE_NAME).
+
+        The MODULE_NAME should be relative to the flask directory.
+
+        Returns:
+            The path to the Flask WSGI application.
+        """
+        return "app:app"
+
+    @property
+    def flask_port(self) -> int:
+        """Gets the port number to use for the Flask server.
+
+        Returns:
+            The port number to use for the Flask server.
+        """
+        return 8000
```

Comment:
This is already defined in a constant

Reply:
This is caused by merging, I will remove the constant, thanks!